src/commands/preload/index.ts

```diff
@@ -161,6 +163,40 @@ Can be repeated to add multiple certificates.\
 			);
 		}
 
+		// Verify that image is not enabled for secure boot. First, confirm it
+		// is a secure boot image with an /opt/*.sig file in the rootA partition.
+		const { explorePartition, BalenaPartition } = await import(
+			'../../utils/image-contents'
+		);
+		const isSecureBoot = await explorePartition<boolean>(
+			params.image,
+			BalenaPartition.ROOTA,
```

Comment:
Not familiar with the `BalenaPartition` abstraction, but what happens here if this is a non flasher image, , with `flash-boot`, `resin-rootA` etc - does it just continue, then get caught on `line 179`?